src/LayerPlan.cpp

```diff
@@ -135,121 +135,52 @@ ExtruderTrain* LayerPlan::getLastPlannedExtruderTrain()
     return last_planned_extruder;
 }
 
-
-Polygons LayerPlan::computeCombBoundaryInside(const size_t max_inset)
+Polygons LayerPlan::computeCombBoundary(const CombBoundary boundary_type)
 {
-    const CombingMode combing_mode = Application::getInstance().current_slice->scene.current_mesh_group->settings.get<CombingMode>("retraction_combing");
-    if (combing_mode == CombingMode::OFF)
+    Polygons comb_boundary;
+    const CombingMode mesh_combing_mode = Application::getInstance().current_slice->scene.current_mesh_group->settings.get<CombingMode>("retraction_combing");
+    if (mesh_combing_mode != CombingMode::OFF && (layer_nr >=0 || mesh_combing_mode != CombingMode::NO_SKIN))
     {
-        return Polygons();
-    }
-    if (layer_nr < 0)
-    { // when a raft is present
-        if (combing_mode == CombingMode::NO_SKIN)
+        if (layer_nr < 0)
         {
-            return Polygons();
+            comb_boundary = storage.raftOutline.offset(MM2INT(0.1));
         }
         else
         {
-            return storage.raftOutline.offset(MM2INT(0.1));
-        }
-    }
-    else 
-    {
-        Polygons comb_boundary;
-        for (const SliceMeshStorage& mesh : storage.meshes)
-        {
-            const SliceLayer& layer = mesh.layers[layer_nr];
-            if (mesh.settings.get<bool>("infill_mesh")) {
-                continue;
-            }
-            const CombingMode combing_mode = mesh.settings.get<CombingMode>("retraction_combing");
-            if (combing_mode == CombingMode::NO_SKIN)
+            for (const SliceMeshStorage& mesh : storage.meshes)
             {
-                // we need to include the walls in the comb boundary otherwise it's not possible to tell if a travel move crosses a skin region
-
-                const coord_t line_width_0 = mesh.settings.get<coord_t>("wall_line_width_0");
+                const SliceLayer& layer = mesh.layers[static_cast<size_t>(layer_nr)];
+                if (mesh.settings.get<bool>("infill_mesh")) // don't process infill_mesh
```

Comment:
Good point. But this could also decrease the print quality of the frist layer if a travel is oozing material on the plate where at a later time a mesh is printed.

Reply:
The function `mesh.isPrinted()` return `true` if the mesh is not infill, cutting or overhang. So it isn't a status if the mesh is already printed and imo a misnomer.

Reply:
That also means that it is allowed to go through the skin of a cutting mesh.

![image](https://user-images.githubusercontent.com/8535734/100890707-d1000b80-34b8-11eb-9bcd-31f4c37cc966.png)
Above picture is with the suggested change. Here a travel can occur through an outside wall when cut with a cutting mesh.

![image](https://user-images.githubusercontent.com/8535734/100890899-060c5e00-34b9-11eb-814c-213355788f47.png)
But changing it to `mesh.settings.get<bool>("infill_mesh") && mesh.settings.get<bool>("anti_overhang_mesh")` gives beter results.

Reply:
Hmm, maybe the cutting mesh is "cropped" to the shape that it cuts? I don't remember exactly how this worked any more.

As long as it doesn't go avoiding the full shape of the cutting mesh (which could be way bigger than the actual normal mesh) then it's fine like this.